front/components/agent_builder/settings/TagsSection.tsx

```diff
@@ -0,0 +1,178 @@
+import { Button, SparklesIcon, useSendNotification } from "@dust-tt/sparkle";
+import { useMemo, useState } from "react";
+import { useController, useFormContext } from "react-hook-form";
+
+import { useAgentBuilderContext } from "@app/components/agent_builder/AgentBuilderContext";
+import type { AgentBuilderFormData } from "@app/components/agent_builder/AgentBuilderFormContext";
+import { TagsSelector } from "@app/components/agent_builder/settings/TagsSelector";
+import { fetchWithErr } from "@app/components/agent_builder/settings/utils";
+import { useTags } from "@app/lib/swr/tags";
+import type {
+  APIError,
+  BuilderSuggestionsType,
+  Result,
+  WorkspaceType,
+} from "@app/types";
+import { isBuilder } from "@app/types";
+import type { TagType } from "@app/types/tag";
+
+async function getTagsSuggestions({
```

Comment:
I would also personally move it to a helper or utils file (again not strongly held, your call)